### PR TITLE
perl*: fix sed reference

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

It was using the shim.

### CODE REVIEW

File: Formula/perl.rb [hunk 1]
```diff
@@ -30,6 +31,7 @@ def install
       -Duseshrplib
       -Duselargefiles
       -Dusethreads
+      -Dsed=/usr/bin/sed
```
Comment (bayandin) [inline Formula/perl.rb:34]: Maybe we need to add `uses_from_macos "sed"` then?

Comment (Bo98) [reply Formula/perl.rb:34]: We do not have a `sed` formula.

This line could be excluded on Linux since we only have a superenv shim for sed on macOS.

Comment (bayandin) [reply Formula/perl.rb:34]: Right, I've confused it with `gnu-sed`. Sorry for noise

Comment (Bo98) [reply Formula/perl.rb:34]: I guess the Linux folk assume system sed exists and is in a consistent location. I'm not entirely sure, but I don't see much `gnu-sed` usage in Linuxbrew/core.

**Comment by Bo98** [comment]
perl@5.18 doesn't appear to build on Catalina:

```
2020-04-16T14:06:37.9981700Z 	Making CPAN::Meta (all)
2020-04-16T14:06:37.9981820Z 
2020-04-16T14:06:37.9983880Z Creating Makefile.PL in cpan/CPAN-Meta for CPAN::Meta
2020-04-16T14:06:37.9984100Z 
2020-04-16T14:06:37.9985130Z Running Makefile.PL in cpan/CPAN-Meta
2020-04-16T14:06:37.9986670Z ../../miniperl -I../../lib Makefile.PL INSTALLDIRS=perl INSTALLMAN1DIR=none INSTALLMAN3DIR=none PERL_CORE=1 LIBPERL_A=libperl.dylib
2020-04-16T14:06:37.9987030Z Writing Makefile for B::Lint
2020-04-16T14:06:37.9987260Z Writing Makefile for CGI
2020-04-16T14:06:37.9990340Z readdir(./../../../../../../..): No such file or directory at /private/tmp/perl@5.18-20200416-40112-5a4hwr/perl-5.18.4/dist/Cwd/lib/File/Spec/Unix.pm line 515.
2020-04-16T14:06:37.9990700Z Writing Makefile for CPAN
2020-04-16T14:06:37.9990920Z /bin/sh: /miniperl: No such file or directory
2020-04-16T14:06:37.9991150Z make[1]: *** [../../lib/auto/CPAN/.exists] Error 127
2020-04-16T14:06:37.9991980Z  /Applications/Xcode.app/Contents/Developer/usr/bin/make config PERL_CORE=1 LIBPERL_A=libperl.dylib failed, continuing anyway...
2020-04-16T14:06:37.9992630Z Making all in cpan/CPAN
2020-04-16T14:06:37.9994390Z  /Applications/Xcode.app/Contents/Developer/usr/bin/make all PERL_CORE=1 LIBPERL_A=libperl.dylib
2020-04-16T14:06:37.9994850Z /bin/sh: /miniperl: No such file or directory
2020-04-16T14:06:37.9995360Z make[1]: *** [../../lib/auto/CPAN/.exists] Error 127
2020-04-16T14:06:37.9997020Z Unsuccessful make(cpan/CPAN): code=512 at make_ext.pl line 490.
2020-04-16T14:06:37.9997330Z make: *** [cpan/CPAN/pm_to_blib] Error 25
```

Though, judging by the bottles, it perhaps never did. Seems to work on Mojave now though.

**Comment by Bo98** [comment]
Paralellism was the problem

**Pull request closed without merging**